lib/api/core/plugins/pluginsManager.js

```diff
@@ -838,13 +839,22 @@ class PluginsManager {
 function triggerHooks(hooks, event, data) {
   const preparedHooks = getPreparedListeners(hooks, event);
 
-  if (preparedHooks) {
-    preparedHooks.forEach(hook => {
-      hook(data);
-    });
-  }
+  return Bluebird.resolve((resolve, reject) => {
+    if (preparedHooks) {
+      preparedHooks.forEach(hook => {
+        try {
+          hook(data);
```

Comment:
this is still broken: plugins aren't the only objects using events

why don't you let the kuzzle object, which inherits from eventemitter, emit that event?